docs/modules/serialization/pages/compact-serialization.adoc

```diff
@@ -326,77 +342,1221 @@ hazelcast:
+        .getCompactSerializationConfig()
+        .addClass(Bar.class);
+----
+--
+
+Java - Client::
++
+--
+[source,java]
+----
+ClientConfig clientConfig = new ClientConfig();
+clientConfig.getSerializationConfig()
+        .getCompactSerializationConfig()
+        .addClass(Bar.class);
+----
+--
+
+.NET::
++
+--
+[source,cs]
+----
+
+----
+--
+
+====
+
+**Declarative Configuration:**
+
+[tabs]
+====
+
+Java - Member - XML::
++
+--
+[source,xml]
+----
+<hazelcast>
+    <serialization>
+        <compact-serialization>
+            <classes>
+                <class>
+                    com.example.Bar
+                </class>
+            </classes>
+        </compact-serialization>
+    </serialization>
+</hazelcast>
+----
+--
+
+Java - Member - YAML::
++
+--
+[source,yaml]
+----
+hazelcast:
+  serialization:
+    compact-serialization:
+      classes:
+        - class: com.example.Bar
+----
+--
+
+Java - Client - XML::
++
+--
+[source,xml]
+----
+<hazelcast-client>
+    <serialization>
+        <compact-serialization>
+            <classes>
+                <class>
+                    com.example.Bar
+                </class>
+            </classes>
+        </compact-serialization>
+    </serialization>
+</hazelcast-client>
+----
+--
+
+Java - Client - YAML::
++
+--
+[source,yaml]
+----
+hazelcast-client:
+  serialization:
+    compact-serialization:
+      classes:
+        - class: com.example.Bar
+----
+--
+
+====
+
+If you want to override the serialization mechanism used for `Serializable` or
+`Externalizable` classes and use Compact serialization without writing any
+serializer in Java, you must add those classes to the configuration.
 
 == GenericRecord Representation
 
-Compact serialized objects
-can also be represented by a `GenericRecord`, without requiring the class in the classpath. See xref:clusters:accessing-domain-objects.adoc[].
+Compact serialized objects can also be represented by a `GenericRecord`, without requiring
+the class in the classpath. See xref:clusters:accessing-domain-objects.adoc[].
+
+== SQL Support
+
+Compact serialized objects can be used in SQL statements, provided that a mappings are created,
+similar to other serialization formats. See xref:sql:mapping-to-maps.adoc[], to learn more
+about Compact serialization mappings.
+
+== WAN Support
+
+Hazelcast supports WAN replication of the Compact serialized objects between different clusters.
+
+However, since the Compact serialization is promoted to the stable status in 5.2, and it is not
+compatible with the previous BETA versions, one has to make sure that the whole WAN cluster
+topology, including all senders and receivers, is at least as new as 5.2, before start replicating data structures containing Compact
+serialized objects.
+
+Since it has promoted to the stable status, it will be possible in the future releases to
+replicate Compact serialized objects between different WAN clusters of versions at least as
+new as 5.2.
```

Comment:
```suggestion
Since it has promoted to the stable status, replicating Compact serialized objects between WAN clusters are of different versions will be possible in the future releases provided that all clusters' versions are at least 5.2.
```

Also I think `at least as new as 5.2` feels long. If you change it, don't forget to change other occurrences.

Reply:
I have removed it from here, but left it in the first paragraph, as I believe it was necessary there